src/shared/previewUtils.ts

```diff
@@ -161,41 +209,47 @@ export class PreviewUtils {
       launchArguments.push({ name: 'LightningExperienceAppID', value: appId });
     }
 
-    launchArguments.push({ name: 'aura.ldpServerUrl', value: ldpServerUrl });
+    launchArguments.push({ name: '0.aura.ldpServerUrl', value: ldpServerUrl });
 
-    launchArguments.push({ name: 'aura.mode', value: auraMode });
+    launchArguments.push({ name: '0.aura.mode', value: auraMode });
 
     return launchArguments;
   }
 
   /**
    * Generates a self-signed certificate and saves it to a file at the specified location.
    *
-   * @param platform A supported platform (Desktop or iOS or Android)
+   * @param platform A mobile platform (iOS or Android)
    * @param saveLocation Path to a folder where the generated certificated will be saved to (defaults to the current working directory)
-   * @returns Path to the generated certificate file
+   * @returns Path to the generated certificate file and the certificate data
    */
-  public static generateSelfSignedCert(platform: Platform, saveLocation = '.'): string {
-    const cert = CryptoUtils.generateSelfSignedCert('localhost', 2048, 365);
+  public static async generateSelfSignedCert(
+    platform: Platform.ios | Platform.android,
+    saveLocation = '.'
+  ): Promise<{ certData: SSLCertificateData; certFilePath: string }> {
+    // See if we have previously generated cert data which is stored in the global config.
+    // If so then use that data otherwise generate new cert data and store it in the global config.
+    let data = await ConfigUtils.getCertData();
+    if (!data) {
+      data = CryptoUtils.generateSelfSignedCert('localhost', 2048, 820);
+      await ConfigUtils.writeCertData(data);
+    }
 
-    // Resolve the save location and ensure it exists
     const basePath = path.resolve(CommonUtils.resolveUserHomePath(saveLocation));
-    if (!fs.existsSync(basePath)) {
-      fs.mkdirSync(basePath);
-    }
 
-    let fileName = '';
-    if (platform === Platform.ios) {
-      // iOS expects the certificate to be a binary DER file
-      fileName = path.join(basePath, 'certificate.der');
-      fs.writeFileSync(fileName, cert.derCertificate, { encoding: 'binary' });
-    } else {
-      // all other platforms can work with a base64 PEM text file
-      fileName = path.join(basePath, 'certificate.pem');
-      fs.writeFileSync(fileName, cert.pemCertificate);
+    const targetFile =
+      platform === Platform.ios ? path.join(basePath, 'localhost.der') : path.join(basePath, 'localhost.pem');
+
+    // If we have not previously generated the cert files then go ahead and do so
+    if (!fs.existsSync(targetFile)) {
+      if (platform === Platform.ios) {
+        fs.writeFileSync(targetFile, data.derCertificate, { encoding: 'binary' });
```

Comment:
So here I'd drop the options specifying encoding, because that's an inherently textual configuration, and doesn't apply to a raw `Buffer`.